pymc/tests/distributions/test_distribution.py

```diff
@@ -0,0 +1,401 @@
+    )
+    def test_density_dist_default_moment_univariate(self, moment, size, expected):
+        if moment == "custom_moment":
+            moment = lambda rv, size, *rv_inputs: 5 * at.ones(size, dtype=rv.dtype)
+        with pm.Model() as model:
+            pm.DensityDist("x", moment=moment, size=size)
+        assert_moment_is_expected(model, expected, check_finite_logp=False)
+
+    @pytest.mark.parametrize("size", [(), (2,), (3, 2)], ids=str)
+    def test_density_dist_custom_moment_univariate(self, size):
+        def density_moment(rv, size, mu):
+            return (at.ones(size) * mu).astype(rv.dtype)
+
+        mu_val = np.array(np.random.normal(loc=2, scale=1)).astype(aesara.config.floatX)
+        with pm.Model():
+            mu = pm.Normal("mu")
+            a = pm.DensityDist("a", mu, moment=density_moment, size=size)
+        evaled_moment = moment(a).eval({mu: mu_val})
+        assert evaled_moment.shape == to_tuple(size)
+        assert np.all(evaled_moment == mu_val)
+
+    @pytest.mark.parametrize("size", [(), (2,), (3, 2)], ids=str)
+    def test_density_dist_custom_moment_multivariate(self, size):
+        def density_moment(rv, size, mu):
+            return (at.ones(size)[..., None] * mu).astype(rv.dtype)
+
+        mu_val = np.random.normal(loc=2, scale=1, size=5).astype(aesara.config.floatX)
+        with pm.Model():
+            mu = pm.Normal("mu", size=5)
+            a = pm.DensityDist(
+                "a", mu, moment=density_moment, ndims_params=[1], ndim_supp=1, size=size
+            )
+        evaled_moment = moment(a).eval({mu: mu_val})
+        assert evaled_moment.shape == to_tuple(size) + (5,)
+        assert np.all(evaled_moment == mu_val)
+
+    @pytest.mark.parametrize(
+        "with_random, size",
+        [
+            (True, ()),
+            (True, (2,)),
+            (True, (3, 2)),
+            (False, ()),
+            (False, (2,)),
+        ],
+    )
+    def test_density_dist_default_moment_multivariate(self, with_random, size):
+        def _random(mu, rng=None, size=None):
+            return rng.normal(mu, scale=1, size=to_tuple(size) + mu.shape)
+
+        if with_random:
+            random = _random
+        else:
+            random = None
+
+        mu_val = np.random.normal(loc=2, scale=1, size=5).astype(aesara.config.floatX)
+        with pm.Model():
+            mu = pm.Normal("mu", size=5)
+            a = pm.DensityDist("a", mu, random=random, ndims_params=[1], ndim_supp=1, size=size)
+        if with_random:
+            evaled_moment = moment(a).eval({mu: mu_val})
+            assert evaled_moment.shape == to_tuple(size) + (5,)
+            assert np.all(evaled_moment == 0)
+        else:
+            with pytest.raises(
+                TypeError,
+                match="Cannot safely infer the size of a multivariate random variable's moment.",
+            ):
+                evaled_moment = moment(a).eval({mu: mu_val})
+
+
+def test_all_distributions_have_moments():
```

Comment:
This seems to be repeated above

Reply:
Still looks unresolved on my end? Codecov is complaining the test above are not being covered, which agress with them still being duplicated.